### PR TITLE
[NVIDIA GPU] Better error message when initializing cudnn

### PR DESCRIPTION
**Pull request by nouiz**
This is to help investigate issues like:
https://github.com/google/jax/issues/15268

### CODE REVIEW

**Review by SandSnip3r — APPROVED** [review]

File: xla/stream_executor/cuda/cuda_dnn.cc [hunk 1]
```diff
@@ -394,10 +394,13 @@ tsl::Status CudnnSupport::Init() {
...
-  if (cerr != cudaSuccess) {
-    LOG(WARNING) << "There was an error before creating cudnn handle: "
-                 << cudaGetErrorName(cerr) << " : " << cudaGetErrorString(cerr);
+  cudaError_t cuda_error = cudaPeekAtLastError();
+  if (cuda_error != cudaSuccess) {
+    // Printing the cuda_error value is useful when cudaGetErrorName doesn't work.
+    LOG(WARNING) << "There was an error before creating cudnn handle ("
+                 << cuda_error << "): " << cudaGetErrorName(cuda_error) << " : "
+                 << cudaGetErrorString(cuda_error);
+    return tsl::Status(absl::StatusCode::kInternal, error);
```
Comment (tpopp) [inline xla/stream_executor/cuda/cuda_dnn.cc:403]: `error` is undefined

**Review by tpopp — APPROVED** [review]

File: xla/stream_executor/cuda/cuda_dnn.cc (hunk 1 above)
Comment (nouiz) [reply xla/stream_executor/cuda/cuda_dnn.cc:405]: I fixed that and added another place with an equivalent error detection mechanism.

**Comment by nouiz** [comment]
One of the CI failed, but I can't see the error.
So I don't know if it is related or not.
Can you take a look?

**Comment by ddunl** [comment]
There were some timeouts, it's trying to submit again right now. I think maybe the timeouts are flakes, hopeful it will get merged on this next try

**Comment by ddunl** [comment]
From what I can tell the only errors are flakes right now, so hopefully I can get this sorted. We run some extra tests on PRs internally now to avoid rollbacks, but for this PR it's only meant extra flakiness! If it isn't resolved tomorrow morning I'll disable those tests and get it merged. Thanks for your PR!!

**Pull request closed without merging**